src/target_properties.cpp

```diff
@@ -73,6 +71,42 @@ std::string inline GetDeviceNameFromMap(const std::string& in)
     return name;
 }
 
-} // namespace miopen
+void TargetProperties::Init(const Handle* const handle)
+{
+    const char* const arch = miopen::GetStringEnv(MIOPEN_DEVICE_ARCH{});
+    if(arch != nullptr && strlen(arch) > 0)
+    {
+        name = arch;
+    }
+    else
+    {
+        name = handle->GetDeviceNameImpl();
+    }
+    name = GetDeviceNameFromMap(name);
 
-#endif // GUARD_MIOPEN_DEVICE_NAME_HPP
+    // Set features to defaults.
+    sramecc = StartsWith(name, "gfx906") || StartsWith(name, "gfx908");
+    xnack   = false;
+    InitDbId();
+}
+
+void TargetProperties::InitDbId()
+{
+    // Let's stay compatible with existing databases:
+    // When feature equal to the default, do not append feature suffice.
+    dbId = name;
+    if(StartsWith(name, "gfx906") || StartsWith(name, "gfx908"))
+    {
+        if(!sramecc)
+            dbId += "_nosramecc";
```

Comment:
@JehandadKhan does this affect Tuna at all?

Reply:
> Also, I think this is a breaking change since perf db and binary cache would miss due to this change.

The PR shouldn't change behavior for the devices we are supporting right now:
- gfx803, xnack off, sramecc off
- gfx900, xnack off, sramecc off
- gfx906, xnack off, sramecc on
- gfx908, xnack off, sramecc on

Reply:
> I believe this needs to be reverted since that implies that the performance parameters as well as the code objects would be different based on these parameters, which should not be the case.

Xnack and sramecc info is present in the CO v4 metadata, which means that CO are different. And it should be so, otherwise HIP/OCL runtime would reject the module.

Switching xnack or sramecc ON and OFF affects either ISA compatibility (correctness) or performance, or both. Therefore, the state of these target features substantially affect the ISA generated by the compiler. Of course, ASM kernels are written taking these target features into account. So far I believe that the simplest approach is considering the same devices with different state of features as _different targets_.

Reply:
Please give it a try and ley me know in case of issues.